second-hand-b/src/main/java/com/codesquad/secondhand/adapter/in/web/config/OAuth2LoginSecurityConfig.java

```diff
@@ -0,0 +1,81 @@
+package com.codesquad.secondhand.adapter.in.web.config;
+
+import static org.springframework.security.config.Customizer.withDefaults;
+
+import com.codesquad.secondhand.adapter.in.web.config.jwt.JwtSignInAuthenticationFilter;
+import com.codesquad.secondhand.adapter.in.web.config.jwt.JwtSignUpAuthenticationFilter;
+import com.codesquad.secondhand.application.port.out.MemberRepository;
+import com.codesquad.secondhand.domain.member.Role;
+import com.codesquad.secondhand.domain.units.JwtTokenProvider;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProviderBuilder;
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository;
+import org.springframework.security.oauth2.client.web.DefaultOAuth2AuthorizedClientManager;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizationRequestRedirectFilter;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.web.SecurityFilterChain;
+
+@RequiredArgsConstructor
+@Configuration
+@EnableWebSecurity
+public class OAuth2LoginSecurityConfig {
+
+    @Autowired
+    JwtTokenProvider jwtTokenProvider;
+
+    @Autowired
+    MemberRepository memberRepository;
+
+    @Bean
+    public SecurityFilterChain filterChain(HttpSecurity httpSecurity) throws Exception {
+        return httpSecurity
+                .authorizeHttpRequests(
+                        requestMatcherRegistry -> requestMatcherRegistry
+                                .mvcMatchers(HttpMethod.GET, "/api/members/signin")
+                                .authenticated()
+                                .mvcMatchers(HttpMethod.POST, "/api/members/signup")
+                                .hasAuthority(Role.USER.getKey())
```

Comment:
```suggestion
                                .mvcMatchers(HttpMethod.GET, "/api/members/signin").authenticated()
                                .mvcMatchers(HttpMethod.POST, "/api/members/signup").hasAuthority(Role.USER.getKey())
```

저희는 회사에서 직관적이여서 이렇게 많이 치곤해요. 요건 스타일적인거라 그냥 고려만 해보셔도 될거같아요